lib/puppet/util/windows/string.rb

```diff
@@ -6,4 +6,12 @@ def wide_string(str)
     str.encode('UTF-16LE')
   end
   module_function :wide_string
+
+  # Read a wide character string up until the first double null, and delete
+  # any remaining null characters.
+  def wstrip(str)
+    str.force_encoding('UTF-16LE').encode('UTF-8', invalid: :replace, undef: :replace).
+      split("\x00")[0].encode(Encoding.default_external)
+  end
```

Comment:
Can we use https://github.com/puppetlabs/puppet/blob/main/lib/puppet/util/windows/api_types.rb#L52 instead of defining this function here?

Reply:
Defined a private method on `Dir` since `read_wide_string` is defined on `FFI::Pointer` instances, and we need it for `FFI::Buffer` instances.